core/src/saros/net/stream/ByteStream.java

```diff
@@ -16,4 +16,12 @@
   public int getReadTimeout() throws IOException;
 
   public void setReadTimeout(int timeout) throws IOException;
+
```

Comment:
The problem is the mighty Packet connection. I do not want to leak the getInput / getOutput stream. And as I can tell you the getId method is already different. I have really a hard time to generalize this stuff as I have to update a bunch of classes and tests. I am not happy with it either.

Reply:
The basic logic goes like this

You call connect on the IConnectionManager. The implementatin goes of through all of it is StreamServices, once a stream is established it will be used by either a PacketConnection or a StreamConnection

Reply:
Ok, I am trying to understand this. This is the basic connection stream interface that is then used to create other connections (of type `IStreamConnection` or `IPacketConnection`). And that is the reason you want to use a different interface for it, even though it has the same signature. Right?

Then this should definitely be mentioned in the `ByteStream` javadoc.

Also, `ByteStream` could be interpreted as the "base stream" as the name suggests that it is only transporting bytes, but the name could still be better to imply that it is the stream used as the basis for all of our connections. Just `BaseStream` would be an option, but its not a huge improvement in my opinion. But maybe just adding the javadoc is enough.